healvis/observatory.py

```diff
@@ -270,6 +287,14 @@ def beam_sq_int(self, freqs, Nside, pointing, beam_pol="pI"):
 
         return beam_sq_int
 
+    def external_beam_val(self, beam, az_arr, za_arr, freqs, pol="XX"):
+        """
+        Call interp() on a beam, and provide results in the right format.
+        """
+        interp_data, interp_basis_vector = beam.interp(az_array=az_arr, za_array=za_arr, 
+                                                        freq_array=freqs)
+        return interp_data[0, 0, 1].T   # just want Npix, Nfreq
```

Comment:
It looks like "pol" isn't used. I don't remember what the interp_data axes are offhand, but you should be careful that the correct polarization is being returned.

Part of the complication with supporting E-field beams it that you have to make sure you're extracting the right feed pair. I suppose the way to do this is (as with the existing code) only support on-diagonal components (XX and YY), and then the "beam_pol" option should be either X or Y.

Reply:
I'll check over this. I haven't been quite sure about the handling of polarization in the sim packages - if they're all doing the same thing or meant to be. Phil is adding polarization to hera_sim so he's grappling with this as well.